### DISCUSSION TITLE
Disable wi-fi ESP32 on request

### DISCUSSION THREAD

**Discussion by lpuntaloro** [HomeSpan/HomeSpan#494] [General]
Hello,
I'm using your library on ESP32 as a bridge and it also handles physical buttons. At night I turn off the home wifi because the router and the bridge are too close to the children's bed. The buttons continue to work normally but I would like ESP not to try to connect by manually breaking the connection, is it possible?
Thank you

**Comment by HomeSpan** [comment]
There is no easy way to do this in HomeSpan at present since HomeSpan will keep trying (and failing) to reconnect to your WiFi network as soon as it finds itself disconnected.

**Comment by lpuntaloro** [comment]
If I interrupt the poll from the main cycle, does he continue anyway?

Luigi Puntaloro
Il 25 gen 2023, 14:12 +0100, HomeSpan ***@***.***>, ha scritto:

**Reply by HomeSpan** [reply]
That would work - all of the action is in homeSpan.poll().  However, your device will no longer respond to button presses either.

**Comment by lpuntaloro** [comment]
In fact I use an external management of the buttons for which it should go.. I try to block the execution on the main loop for a while and let's see what happens...

Luigi Puntaloro
Il 25 gen 2023, 14:26 +0100, HomeSpan ***@***.***>, ha scritto:
